packages/sdks/snippets/react-sdk-next-pages/src/components/CustomTabs.tsx

```diff
@@ -0,0 +1,63 @@
+// components/CustomTabs.tsx
+'use client';
+import type {
+  BuilderBlock,
+  RegisteredComponent,
+  RegisteredComponents,
+} from '@builder.io/sdk-react';
+import { Blocks } from '@builder.io/sdk-react';
+import { useState } from 'react';
+
+type CustomTabsProps = {
+  tabList: {
+    tabName: string;
+    blocks: BuilderBlock[];
+  }[];
+  builderBlock: BuilderBlock;
+  builderComponents: RegisteredComponents;
+};
+
+export function CustomTabs({
+  tabList,
+  builderBlock,
+  builderComponents,
+}: CustomTabsProps) {
+  const [activeTab, setActiveTab] = useState(0);
+
+  return (
+    <>
+      {tabList.map((tab, index) => (
```

Comment:
can we add optional chaining? I think this works for us because we do have `tabList` but if there are no default items then it will throw us an error

```suggestion
      {tabList?.map((tab, index) => (
```